File: Backend/src/main/java/org/example/backend/dto/GradeDto.java
package org.example.backend.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Value;

import java.io.Serializable;
import java.time.Instant;

/**
 * DTO for {@link org.example.backend.domain.Grade}
 */
@Data
@NoArgsConstructor
public class GradeDto implements Serializable {
    Integer id;
    Short value;
    String type;
    String comment;
    Instant dateRecorded;
    StudentDto student;
    TimeslotDto timeslot;
}

File: Frontend/src/app/dto/grade-dto.ts
import {StudentDto} from "./student-dto";
import {TimeslotDto} from "./timeslot-dto";

export interface GradeDto {
  id : number
  value : number
  type : string
  comment : string
  dateRecorded : Date
  student : StudentDto
  timeslot : TimeslotDto
}
